src/Razor/test/Microsoft.NET.Sdk.Razor.Test/IntegrationTests/StaticAssetsIntegrationTest.cs

```diff
@@ -0,0 +1,296 @@
+            Output = output;
+        }
+
+        public PackageTestProjectsFixture PackageTestProjects { get; private set; }
+
+        public ITestOutputHelper Output { get; private set; }
+
+        [Fact]
+        [InitializeTestProject("AppWithPackageAndP2PReference")]
+        public async Task Build_GeneratesAspNetCoreStaticAssetsManifest_Success_CreatesManifest()
+        {
+            // For some reason when using a custom package cache the imports won't get added on
+            // the initial restore, so we restore the packages ourselves.
+            await DotnetMSBuild("Restore");
+
+            var expectedManifest = GetExpectedManifest();
+
+            var result = await DotnetMSBuild("Build");
+
+            Assert.BuildPassed(result);
+
+            // GenerateAspNetCoreStaticAssetsManifest should generate the manifest and the cache.
+            Assert.FileExists(result, IntermediateOutputPath, "Microsoft.AspNetCore.StaticAssets.xml");
+            Assert.FileExists(result, IntermediateOutputPath, "Microsoft.AspNetCore.StaticAssets.cache");
+
+            var path = Assert.FileExists(result, OutputPath, "AppWithPackageAndP2PReference.dll");
+            var assembly = Assert.ContainsEmbeddedResource(path, "Microsoft.AspNetCore.StaticAssets.xml");
+            using (var reader = new StreamReader(assembly))
+            {
+                var data = await reader.ReadToEndAsync();
+                Assert.Equal(expectedManifest, data);
+            }
+        }
+
+        private string GetExpectedManifest()
+        {
+            var projectNames = PackageTestProjectsFixture.GetProjectsToPack()
+                .Select(p => Path.GetFileNameWithoutExtension(p).ToLowerInvariant())
+                .OrderByDescending(p => p.Length)
+                .ToArray();
+
+            var restorePath = LocalNugetPackagesCacheTempPath;
+            var projects = projectNames.Select(p =>
+                Path.Combine(
+                    restorePath,
+                    p,
+                    "1.0.0",
+                    p.Contains("transitive", StringComparison.OrdinalIgnoreCase) ? "buildTransitive" : "build", "..", "razorContent") + Path.DirectorySeparatorChar)
+                .ToArray();
+
+            return $@"<AspNetCoreStaticAssets Version=""1.0"">
+  <ContentRoot BasePath=""_content/PackageLibraryTransitiveDependency"" Path=""{projects[0]}"" />
+  <ContentRoot BasePath=""_content/PackageLibraryDirectDependency"" Path=""{projects[1]}"" />
+</AspNetCoreStaticAssets>";
+        }
+
+        [Fact]
+        [InitializeTestProject("SimpleMvc")]
+        public async Task Build_DoesNotEmbedManifestWhen_NoStaticResourcesAvailable()
+        {
+            // For some reason when using a custom package cache the imports won't get added on
+            // the initial restore, so we restore the packages ourselves.
+            await DotnetMSBuild("Restore");
+
+            var result = await DotnetMSBuild("Build");
```

Comment:
You should be able to do `await DonetMsbuild("Build", "/restore");` instead. Same above.

Reply:
changed